fabric/src/main/java/me/shedaniel/architectury/registry/fabric/TradeRegistryImpl.java

```diff
@@ -0,0 +1,31 @@
+package me.shedaniel.architectury.registry.fabric;
+
+import net.minecraft.world.entity.npc.VillagerProfession;
+import net.minecraft.world.entity.npc.VillagerTrades;
+import org.apache.commons.lang3.ArrayUtils;
+import org.apache.commons.lang3.math.NumberUtils;
+
+public class TradeRegistryImpl {
```

Comment:
This needs to use the fabric trade registry, or else it will unregister on their own (Not our fault technically, their api is really really awful)

Reply:
I didn't know that it existed, but I changed it